src/main/java/com/asyncy/apis/hub/entities/Repo.java

```diff
@@ -0,0 +1,80 @@
+package com.asyncy.apis.hub.entities;
+
+
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import org.json.JSONObject;
+import org.springframework.web.util.UriTemplate;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Repo {
+    private final int id;
+    private final String name;
+    private final String full_name;
+    private final String owner;
+    private final String description;
+    private final UriTemplate contentsUrl;
+    private final Boolean _private;
+    private Boolean valid = false;
+
+    public Repo(int id, String name, String full_name, String description, String contents_url, String owner, Boolean _private) {
+        this.id = id;
+        this.name = name;
+        this.full_name = full_name;
+        this.description = description;
+        this.contentsUrl = new UriTemplate(contents_url);
+        this._private = _private;
+        this.owner = owner;
+    }
+
+    public Repo(JSONObject repo) {
+        this.id = repo.getInt("id");
+        this.name = repo.getString("name");
+        this.full_name = repo.getString("full_name");
+        this.description = repo.optString("description", "");
+        this.contentsUrl = new UriTemplate(repo.getString("contents_url"));
+        this._private = repo.getBoolean("private");
+        this.owner = repo.getJSONObject("owner").getString("login");
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getFull_name() {
+        return full_name;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getOwner() {
+        return owner;
+    }
+
+    public Boolean isPrivate() {
```

Comment:
Use unboxed boolean.
```suggestion
     public boolean isPrivate() {
```